lib/leaderboard/denokv/denokv_leaderboard_client.ts

```diff
@@ -118,122 +132,62 @@ export class DenoKvLeaderboardClient implements LeaderboardClient {
     return { ok: true };
   }
 
-  public async submit(
-    discord_user_id: string,
-    lc_submission_id: string,
-    currentDate = new Date(),
-  ): Promise<api.SubmitResponse> {
-    // Check if the player is registered in our leaderboard.
-    const maybePlayerResult = await this.kv
-      .get<api.LCPlayer>([LeaderboardKvPrefix.PLAYERS, discord_user_id]);
-    if (!maybePlayerResult.value) {
-      throw new Error("Player not registered");
-    }
-
-    // Find the accepted submission in the recent submissions of the player.
-    const recentAcceptedSubmissions = await this.lc
-      .getRecentAcceptedSubmissions(maybePlayerResult.value.lc_username, 10);
-    const acceptedSubmission = recentAcceptedSubmissions
-      .find((s) => s.id === lc_submission_id);
-    if (!acceptedSubmission) {
-      throw new Error("Submission not found");
-    }
-
-    // The latest season. Default to empty season when not found or not latest.
-    // If current date is no longer in the "latest" season, create a new season.
-    const maybeSeasonResult = await this.getLatestSeasonFromKv();
-    const isLatestSeason = !!(maybeSeasonResult?.value) && checkDateInWeek(
-      new Date(maybeSeasonResult.value.start_date).getTime(),
-      currentDate.getTime(),
+  public async sync(
+    seasonID?: string,
+    referenceDate = new Date(),
+  ): Promise<api.SyncResponse> {
+    // startOfWeekUTC is the start of the season in UTC.
+    const startOfWeekUTC = getStartOfWeek(this.restartMs, referenceDate);
+
+    // Get the season.
+    let season: api.Season;
+    let seasonResult: Deno.KvEntryMaybe<api.Season> | null = null;
+    if (seasonID) {
+      seasonResult = await this.kv.get<api.Season>([
+        LeaderboardKvPrefix.SEASONS,
+        seasonID,
+      ]);
+      if (!seasonResult.value) {
+        throw new Error("Season not found");
+      }
+
+      season = seasonResult.value;
+    } else {
+      seasonResult = await this.getLatestSeasonFromKv();
+      season = seasonResult?.value
```

Comment:
Set `season` to the return value of `makeEmptySeason` if `seasonResult?.value === undefined` OR the most recently synced season is not `checkDateInWeek` of `startOfWeekUTC` and `seasonResult.value.start_date`.

Reply:
Resolved in <https://github.com/acmcsufoss/lc-dailies/pull/41/commits/2abcf63b35feec1b5cd0a663712c2a4d0a9a6f4f>.